modules/gapi/src/streaming/onevpl/utils.cpp

```diff
@@ -429,4 +432,69 @@ std::string ext_mem_frame_type_to_cstr(int type) {
 } // namespace wip
 } // namespace gapi
 } // namespace cv
+
+namespace {
+namespace cfg {
+// FIXME: Move OneVPL arguments parser to a single place
+typename cv::gapi::wip::onevpl::CfgParam create_from_string(const std::string &line) {
+    using namespace cv::gapi::wip;
+
+    if (line.empty()) {
+        throw std::runtime_error("Cannot parse CfgParam from emply line");
+    }
+
+    std::string::size_type name_endline_pos = line.find(':');
+    if (name_endline_pos == std::string::npos) {
+        throw std::runtime_error("Cannot parse CfgParam from: " + line +
+                                 "\nExpected separator \":\"");
+    }
+
+    std::string name = line.substr(0, name_endline_pos);
+    std::string value = line.substr(name_endline_pos + 1);
+
+    return cv::gapi::wip::onevpl::CfgParam::create(name, value,
+                                                   /* vpp params strongly optional */
+                                                   name.find("vpp.") == std::string::npos);
+}
+} // namespace cfg
+} // anonymous namespace
+
+namespace cv {
+namespace gapi {
+namespace wip {
+namespace onevpl {
+
+cv::Ptr<cv::gapi::wip::IStreamSource> make_default_onevpl_src(const std::string& input) {
```

Comment:
What if I have not gpu at all (or DX11)? Why is it chosen default?
It brings up too much questions. This function looks unstable, and definitely must not be exist as interface in library utils

Reply:
Tend to agree, as this function is named "default", it should work universally in all possible circumstances (e.g. fallback on CPU, etc).

Reply:
Gonna add 1 more argument for platform selection

Reply:
I disagree: it's not a better solution to implement a functions which has only one purpose to take one/two arguments and call make_onevpl_src. I mean - it wraps only 3 rows and it  seems useless.

I insist to remove this function and use exactly the same 3 rows in your samples directly